### PR TITLE
feat: new approach to pass execution config by file

### PR DESCRIPTION
**Pull request by StarpTech**
## Motivation and Context

```yaml
execution_config:
  file:
    path: "router.json" # EXECUTION_CONFIG_FILE_PATH
    watch: true # Possible future addition
```

`router_config_path` has been deprecated.

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: router/pkg/config/config.schema.json [hunk 1]
```diff
@@ -1211,7 +1235,7 @@
     "router_config_path": {
       "type": "string",
       "format": "file-path",
-      "deprecationMessage": "The router_config_path is deprecated. Please use the execution_config instead.",
+      "deprecationMessage": "The router_config_path is deprecated. Please use the execution_config.local instead.",
```
Comment (JivusAyrus) [inline router/pkg/config/config.schema.json:1238]: execution_config.file.path?

**Review by JivusAyrus — CHANGES_REQUESTED** [review]

File: router/pkg/config/config.schema.json (hunk 1 above)
Comment (StarpTech) [reply router/pkg/config/config.schema.json:1238]: Good eyes.

File: router/pkg/config/config.go [hunk 2]
```diff
@@ -451,13 +451,18 @@ type PersistedOperationsCDNProvider struct {
...
 }
 
-type ExecutionConfigStorageConfig struct {
+type ExecutionConfigStorage struct {
 	ProviderID string `yaml:"provider_id,omitempty" env:"EXECUTION_CONFIG_STORAGE_PROVIDER_ID"`
 	ObjectPath string `yaml:"object_path,omitempty" env:"EXECUTION_CONFIG_STORAGE_OBJECT_PATH"`
 }
 
+type ExecutionConfigFile struct {
+	Path string `yaml:"path,omitempty" env:"EXECUTION_CONFIG_FILE_PATH"`
```
Comment (jensneuse) [inline router/pkg/config/config.go:460]: Is watch missing here?

**Review by jensneuse — APPROVED** [review]

File: router/pkg/config/config.go (hunk 2 above)
Comment (StarpTech) [reply router/pkg/config/config.go:460]: No, see comment. It is just a proposal how this interface allows for extensions.

**Review by JivusAyrus — APPROVED** [review]

**Pull request merged**